src/Solver.cpp

```diff
@@ -371,6 +390,51 @@ void Solver::useExponentialIntrinsic() {
 
 
 /**
+ * @brief Initializes the FSR volumes and Materials array.
+ * @details This method assigns each FSR a unique, monotonically increasing
+ *          ID, sets the Material for each FSR, and assigns a volume based on
+ *          the cumulative length of all of the segments inside the FSR.
+ */
+void Solver::initializeFSRs() {
+
+  log_printf(INFO, "Initializing flat source regions...");
+
+  /* Delete old FSR arrays if they exist */
+  if (_FSR_volumes != NULL)
+    delete [] _FSR_volumes;
+
+  if (_FSR_materials != NULL)
+    delete [] _FSR_materials;
+
+  /* Get an array of volumes indexed by FSR  */
+  _FSR_volumes = _track_generator->getFSRVolumes();
+
+  /* Allocate an array of Material pointers indexed by FSR */
+  _FSR_materials = new Material*[_num_FSRs];
+
+  /* Compute the number of fissionable Materials */
+  std::map<int, Material*> all_materials = _geometry->getAllMaterials();
```

Comment:
`all_materials` doesn't seem to be used anywhere so I would recommend taking this line out.